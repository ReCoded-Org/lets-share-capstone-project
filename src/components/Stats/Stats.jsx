import Image from "next/image";
import { useTranslation } from "next-i18next";
import { MdSwitchAccount } from "react-icons/md";
import { FaChartLine } from "react-icons/fa";
import { BsFillPinMapFill } from "react-icons/bs";
import { MdRecommend } from "react-icons/md";

function Stats() {
    const { t } = useTranslation("common");
    return (
        <div className='bg-[#FAFAFA] font-primary'>
            <div className='flex  h-60 justify-center bg-primary md:h-80 xl:h-80'>
                <Image
                    src='/images/donate.png'
                    alt='donate'
                    width={600}
                    height={1000}
                    className=' -z-1 ml-10 h-full w-full opacity-25'
                />
            </div>
            <div className='container relative mx-auto -mt-12   grid  grid-cols-1 gap-x-6 gap-y-6 rounded-2xl bg-[white] px-4 py-8  shadow-lg sm:grid-cols-2 md:-mt-24 lg:grid-cols-4 xl:-mt-24'>
                <div className='group flex basis-1/4 flex-col items-center gap-3 rounded-lg p-3 text-center transition-all duration-300 hover:-translate-y-3 hover:bg-[#f3f3f3] hover:shadow-lg'>
                    <MdSwitchAccount color='#F07167' size={60} />
                    <h2 className='font-bold text-fontColor'>
                        {t("landing.stats.accountsTitle")}
                    </h2>
                    <p className='text-fontColor '>
                        {t("landing.stats.accounts")}
                    </p>
                </div>

                <div className='group flex basis-1/4 flex-col items-center gap-3 rounded-lg p-3 text-center transition-all duration-300 hover:-translate-y-3 hover:bg-[#f3f3f3] hover:shadow-lg'>
                    <FaChartLine color='#F07167' size={60} />
                    <h2 className='font-bold text-fontColor'>
                        {t("landing.stats.beneficiariesTitle")}
                    </h2>
                    <p className='text-fontColor'>
                        {t("landing.stats.beneficiaries")}
                    </p>
                </div>

                <div className='group flex basis-1/4 flex-col items-center gap-3 rounded-lg p-3 text-center transition-all duration-300 hover:-translate-y-3 hover:bg-[#f3f3f3] hover:shadow-lg'>
                    <BsFillPinMapFill color='#F07167' size={60} />
                    <h2 className='font-bold text-fontColor'>
                        {t("landing.stats.GeographicalDistrictsTitle")}
                    </h2>
                    <p className='text-fontColor '>
                        {t("landing.stats.GeographicalDistricts")}
                    </p>
                </div>

                <div className='group flex basis-1/4 flex-col items-center gap-3 rounded-lg p-3 text-center transition-all duration-300 hover:-translate-y-3 hover:bg-[#f3f3f3] hover:shadow-lg'>
                    <MdRecommend color='#F07167' size={60} />
                    <h2 className='font-bold text-fontColor'>
                        {t("landing.stats.recommendationsTitle")}
                    </h2>
                    <p className='text-fontColor '>
                        {t("landing.stats.recommendations")}
                    </p>
                </div>
            </div>
        </div>
    );
}

export default Stats;
