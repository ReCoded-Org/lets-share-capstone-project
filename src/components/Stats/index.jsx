import Stats from "./Stats.jsx";

export default Stats;
