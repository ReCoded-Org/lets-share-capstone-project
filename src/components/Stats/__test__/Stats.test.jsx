"use strict";

import renderer from "react-test-renderer";

import Stats from "../Stats";

it("renders correctly", () => {
    const tree = renderer.create(<Stats />).toJSON();
    expect(tree).toMatchSnapshot();
});
