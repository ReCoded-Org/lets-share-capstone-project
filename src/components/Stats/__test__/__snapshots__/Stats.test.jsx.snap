// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="flex h-60 justify-center bg-primary md:h-80 xl:h-80"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%271000%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="donate"
        className=" -z-1 ml-10 h-full w-full opacity-25"
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="donate"
          className=" -z-1 ml-10 h-full w-full opacity-25"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=1200&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1200&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>,
  <div
    className="relative ml-8 mr-8 -mt-12 grid grid-cols-1 items-center gap-x-6 gap-y-6 rounded-2xl bg-[white] py-8 px-2 shadow-lg sm:grid-cols-2 md:-mt-24 lg:grid-cols-4 xl:-mt-24"
  >
    <div
      className="basis-1/4 text-center "
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="icon"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="icon"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <h2
        className="font-bold text-fontColor "
      >
        1/3
      </h2>
      <p
        className="text-fontColor"
      >
        Lorem ipsum dolor sit amet,
        <br />
         consectetur adipiscing elit,
        <br />
         sed do eiusmod tempor incididunt ut
        <br />
         labore et dolore magna aliqua.
      </p>
    </div>
    <div
      className="basis-1/4 text-center "
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="icon"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="icon"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <h2
        className="font-bold text-fontColor "
      >
        75%
      </h2>
      <p
        className="text-fontColor"
      >
        Lorem ipsum dolor sit amet,
        <br />
         consectetur adipiscing elit,
        <br />
         sed do eiusmod tempor incididunt ut
        <br />
         labore et dolore magna aliqua.
      </p>
    </div>
    <div
      className="basis-1/4 text-center "
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="icon"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="icon"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <h2
        className="font-bold text-fontColor "
      >
        300
      </h2>
      <p
        className="text-fontColor"
      >
        Lorem ipsum dolor sit amet,
        <br />
         consectetur adipiscing elit,
        <br />
         sed do eiusmod tempor incididunt ut
        <br />
         labore et dolore magna aliqua.
      </p>
    </div>
    <div
      className="basis-1/4 text-center "
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="icon"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="icon"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <h2
        className="font-bold text-fontColor "
      >
        Facts
      </h2>
      <p
        className="text-fontColor"
      >
        Lorem ipsum dolor sit amet,
        <br />
         consectetur adipiscing elit,
        <br />
         sed do eiusmod tempor incididunt ut
        <br />
         labore et dolore magna aliqua.
      </p>
    </div>
  </div>,
]
`;
